sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -851,14 +853,16 @@ object StateStore extends Logging {
       threadPool.shutdown() // Disable new tasks from being submitted
 
       // Wait a while for existing tasks to terminate
-      if (!threadPool.awaitTermination(5 * 60, TimeUnit.SECONDS)) {
+      val waitTimeout = 5 * shutdownTimeout
```

Comment:
Wait, since the config is control the timeout to wait for shutdown, shouldn't the config refer to the `waitTimeout` here?

Reply:
Done - updated